apps/voting-disputable/contracts/DisputableVoting.sol

```diff
@@ -0,0 +1,803 @@
+        initialized();
+
+        require(isContract(_token), ERROR_TOKEN_NOT_CONTRACT);
+        require(_voteTime > 0, ERROR_VOTE_TIME_ZERO);
+
+        token = _token;
+        voteTime = _voteTime;
+
+        _changeSupportRequiredPct(_supportRequiredPct);
+        _changeMinAcceptQuorumPct(_minAcceptQuorumPct);
+        _changeOverruleWindow(_overruleWindow);
+    }
+
+    /**
+    * @notice Change required support to `@formatPct(_supportRequiredPct)`%
+    * @param _supportRequiredPct New required support
+    */
+    function changeSupportRequiredPct(uint64 _supportRequiredPct)
+        external
+        authP(MODIFY_SUPPORT_ROLE, arr(uint256(_supportRequiredPct), uint256(supportRequiredPct)))
+    {
+        _changeSupportRequiredPct(_supportRequiredPct);
+    }
+
+    /**
+    * @notice Change minimum acceptance quorum to `@formatPct(_minAcceptQuorumPct)`%
+    * @param _minAcceptQuorumPct New acceptance quorum
+    */
+    function changeMinAcceptQuorumPct(uint64 _minAcceptQuorumPct)
+        external
+        authP(MODIFY_QUORUM_ROLE, arr(uint256(_minAcceptQuorumPct), uint256(minAcceptQuorumPct)))
+    {
+        _changeMinAcceptQuorumPct(_minAcceptQuorumPct);
+    }
+
+    /**
+    * @notice Change overrule window to `@transformTime(_overruleWindow)`
+    * @param _overruleWindow New overrule window in seconds
+    */
+    function changeOverruleWindow(uint64 _overruleWindow)
+        external
+        authP(MODIFY_OVERRULE_WINDOW_ROLE, arr(uint256(_overruleWindow), uint256(overruleWindow)))
+    {
+        _changeOverruleWindow(_overruleWindow);
+    }
+
+    /**
+    * @notice `_representative == 0x0 ? 'Set your voting representative to ' + _representative : 'Remove your representative'`
+    * @param _representative Address of the representative who is allowed to vote on behalf of the sender. Use the zero address for none.
+    */
+    function setRepresentative(address _representative) external isInitialized {
+        representatives[msg.sender] = _representative;
+        emit ChangeRepresentative(msg.sender, _representative);
+    }
+
+    /**
+    * @notice Create a new vote about "`_context`"
+    * @param _executionScript EVM script to be executed on approval
+    * @param _context Vote context
+    * @return voteId Id for newly created vote
+    */
+    function newVote(bytes _executionScript, bytes _context) external auth(CREATE_VOTES_ROLE) returns (uint256 voteId) {
+        return _newVote(_executionScript, _context);
+    }
+
+    /**
+    * @notice Vote `_supports ? 'yes' : 'no'` in vote #`_voteId`
+    * @dev Initialization check is implicitly provided by `voteExists()` as new votes can only be
+    *      created via `newVote()`, which requires initialization
+    * @param _voteId Id for vote
+    * @param _supports Whether voter supports the vote
+    */
+    function vote(uint256 _voteId, bool _supports) external voteExists(_voteId) {
+        require(_canVote(votes[_voteId], msg.sender), ERROR_CANNOT_VOTE);
```

Comment:
We might be able to move this check into `_vote()` now—it will also save an extra SLOAD :)